protollm/connectors/rest_server.py

```diff
@@ -0,0 +1,113 @@
+import json
+from typing import Any, Dict, List, Optional, Union
+
+import requests
+from langchain_core.callbacks import CallbackManagerForLLMRun
+from langchain_core.language_models import BaseChatModel
+from langchain_core.messages import (AIMessage, BaseMessage, HumanMessage,
+                                     SystemMessage)
+from langchain_core.outputs import ChatGeneration, ChatResult
+
+
+class ChatRESTServer(BaseChatModel):
+    model: Optional[str] = 'llama3'
+    base_url: str = 'http://10.32.2.2:8672'
+    timeout: Optional[int] = None
+    """Timeout for the request stream"""
+
+    @property
+    def _llm_type(self) -> str:
+        """Return type of chat model."""
+        return "chat-rest-server"
+
+    def _convert_messages_to_rest_server_messages(
+            self, messages: List[BaseMessage]
+    ) -> List[Dict[str, Union[str, List[str]]]]:
+        chat_messages: List = []
+        for message in messages:
+            role = ""
+            if isinstance(message, HumanMessage):
+                role = "user"
+            elif isinstance(message, AIMessage):
+                role = "assistant"
+            elif isinstance(message, SystemMessage):
+                role = "system"
+            else:
+                raise ValueError(
+                    "Received unsupported message type.")
+
+            content = ""
+            if isinstance(message.content, str):
+                content = message.content
+            else:
+                raise ValueError(
+                    "Unsupported message content type. "
+                    "Must have type 'text' "
+                )
+            chat_messages.append(
+                {
+                    "role": role,
+                    "content": content
+                }
+            )
+        return chat_messages
+
+    def _create_chat(
+            self,
+            messages: List[BaseMessage],
+            stop: Optional[List[str]] = None,
+            **kwargs: Any,
+    ) -> Dict:
```

Comment:
As I understood, that would be a little better

```suggestion
    ) -> Dict[str, Any]:
```